File: src/main/java/com/itsci/mju/maebanjumpen/model/Penalty.java
package com.itsci.mju.maebanjumpen.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Entity
@Table(name="penalty")
@AllArgsConstructor
@NoArgsConstructor
@Data
@JsonIgnoreProperties({"hibernateLazyInitializer","handler"})
public class Penalty {
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Integer penaltyId;
	@Column(nullable = false)
	private String penaltyType;
	@Column(nullable = false)
	private String penaltyDetail;
	@Column(nullable = false)
	private LocalDateTime penaltyDate;
	@Column(nullable = false)
	private String penaltyStatus;


	@OneToOne(mappedBy = "penalty", fetch = FetchType.LAZY)
	private Report report;
}

File: src/main/kotlin/com/itsci/mju/maebanjumpen/entity/Penalty.kt
package com.itsci.mju.maebanjumpen.entity

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import jakarta.persistence.*
import java.time.LocalDateTime

@Entity
@Table(name = "penalty")
@JsonIgnoreProperties("hibernateLazyInitializer", "handler")
data class Penalty(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = 0,

    @Column(nullable = false)
    var penaltyType: String = "",

    @Column(nullable = false)
    var penaltyDetail: String = "",

    @Column(nullable = false)
    var penaltyDate: LocalDateTime? = null,

    @Column(nullable = false)
    var penaltyStatus: String = "",

    @OneToOne(mappedBy = "penalty", fetch = FetchType.LAZY)
    var report: Report? = null
) {
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is Penalty) return false
        return id != null && id == other.id
    }

    override fun hashCode(): Int = id?.hashCode() ?: 0

    override fun toString(): String = "Penalty(penaltyId=$id, penaltyType='$penaltyType', penaltyStatus='$penaltyStatus')"
}

